### DISCUSSION TITLE
BERTopic Topic Aspects

### DISCUSSION THREAD

**Discussion by rizkiamandaputri** [MaartenGr/BERTopic#1428] [Q&A] [answered]
'BERTopic' object has no attribute 'topic_aspects_'

How to slove this?

**Comment by MaartenGr** [comment]
Please provide a bit more context. The code, version of BERTopic and its dependencies, etc.

**Reply by rizkiamandaputri** [reply]
Thank you before for answer my question.

I tried this code to label the terms :

{topic: " | ".join(list(zip(*values))[0][:3]) for topic, values in topic_model.topic_aspects_["KeyBERT"].items()}
topic_model.set_topic_labels(keybert_topic_labels)

But the result is 'BERTopic' object has no attribute 'topic_aspects_'

How to slove this?

**Reply by MaartenGr** [reply]
You might have an earlier version of BERTopic. Updating to the latest version might help.

**Reply by rizkiamandaputri** [reply]
Thank you Maarten for reply my comment
I have tried updating to the latest version of BERTopic, but I got new another error like this hahaha

![Command Prompt - py  -m uvicorn main_app --reload --port 6565 26_07_2023 22_53_23](https://github.com/MaartenGr/BERTopic/assets/66537975/d710f48e-6d9a-4748-b59d-7dd18270737c)

Is my code program wrong? Need your help

**Reply by MaartenGr** [reply]
Could you share your full code from beginning to end? There might be something going on with initialization or perhaps somewhere else. Getting the complete picture would help.

**Reply by rizkiamandaputri** [reply]
Sorry for taking your time. The picture of the full code are here :
https://drive.google.com/file/d/1jFJfAhwrM4jc0g5OjKFzsvdUWei3dKu6/view?usp=drive_link

Thank you Maarten for answer my problem

**Reply by MaartenGr** [reply]
I can't access the file and just sent you a request to do so.

**Reply by MaartenGr** [reply]
It seems you shared a zip file. I generally do not open these kinds of files on my system. Could you instead share your code here? I am mostly interested in how you initialized BERTopic.

**Reply by rizkiamandaputri** [reply]
Sorry for taking your time. You can check my github repository here : https://github.com/rizkiamandaputri/BERTopic.git
Thank you so much Maarten

**Reply by MaartenGr** [reply]
It shows a 404 so I cannot seem to access the code. You can also copy-past your code in the comments here instead.

**Reply by rizkiamandaputri** [reply]
Hi Maarten
Sorry for taking your time. I created the file into private so you couldn't open it. And now, you can check my full code on the same link : https://github.com/rizkiamandaputri/BERTopic.git . Thank you so much Maarten for your help

**Reply by MaartenGr** [reply, marked as answer]
When you want to use `KeyBERTInspired`, you should actually pass it to the topic model during training. That is why you are getting that error. I would advise reading through the guide [here](https://maartengr.github.io/BERTopic/getting_started/representation/representation.html#keybertinspired).

**Reply by rizkiamandaputri** [reply]
Thank you so much Maarten
Your advise is really help me to understand BERTopic better
